### DISCUSSION TITLE
rAudio • 20220909: No more nested folder structure for webradios possible

### DISCUSSION THREAD

**Discussion by klassikmann** [rern/rAudio#1021] [Fixed Issues]
Some of my webradios are organized using a two-step directure structure.
Example:
```
ARD/
├── BR
├── HR
├── MDR
├── NDR
├── RB
├── RBB
├── SR
├── SWR
└── WDR
```
(ARD is a joint organisation of Germany's regional public-service broadcasters, and BR, HR, ... are the state broadcasting corporations.)

The channels of the various broadcasters are located in the second directory level. The structure had been created locally, i.e. outside of the rAudio GUI.

Now rAudio shows empty directories for the bottom folders in the webradio view. 

Workaround: Move the bottom directories to the top. But this makes the webradio view very overcrowded.

Request: Allow nested folders to organize the webradios as before.

**Comment by rern** [comment]
Fixed with today's update.

**Reply by klassikmann** [reply]
> Fixed with today's update.

Hm? 

1. I moved the webradios of the regional public-service broadcasters (BR, HR, ...) from the top-level directory back to the ARD directory.
2. I applied today's update.
3. Reboot.
4. If I select any of my top-level webradio folders (any, not ARD only!) I get the webradios of my alphabetically last webradio folder, nothing else.
5. I started a "Library Database update -> Changed files, only" which never ended. I killed the `find -L -type f ! -path */img/*` process after 50 minutes 100% CPU time. But the process has been restarted.
6. Reboot. Redo update.
7. Back to start: I moved the sub-folders BR, HR, ... back to the top.

What I have done wrong?

**Reply by klassikmann** [reply]
Add-on:
Now I tried a full library update. Observations:

1. The `mpc update` part seems to finish successfully.
2. The "latest" file points to a very old entry.
3. The `find -L -type f ! -path */img/*` - executed by cmd-list.sh to count the number of dabradios (not used here) and webradios - runs again "until doomsday". 
4. Any webradio folder shows the contents of the same webradio folder. Now a different one than before, not the alphabetically last webradio folder. Webradios are unusable now.

**Comment by AndyFx12** [comment]
Same here. Only toplevel radios work.

**Reply by rern** [reply]
@klassikmann, @AndyFx12,

Please try the UPDATE branch.

**Reply by AndyFx12** [reply]
Update works fine with webradios - even with sub-sub-folders. 
Thanks!

*Update* 
Library-update runs "forever". See klassikmann´s comment about "find ..." Had to kill manually.
![grafik](https://user-images.githubusercontent.com/35779271/190844587-353f69db-fb52-421e-a1b3-e568254c7cc5.png)

**Reply by klassikmann** [reply]
1. The problems regarding webradios are solved.
2. The problem regarding "endless" library update persists. `mpc update` is finishing correctly, but the subsequent `find -L -type f ! -path */img/*` runs "endless" with 100% CPU usage.
Observation 1: **All variants of library updates, global but also local updates show the same effect. This prevents managing the library.**
Observation 2: Now, all files in `/srv/http/data/mpd` are owned by mpd:audio. In the past all files except of mdb.db were owned by http:http.

**Reply by bTronics** [reply]
How can I manually kill the 'endless' library update? (which command?)

**Reply by rern** [reply]
@AndyFx12, @klassikmann,

Please try the UPDATE branch again for the `find` issue. (Fix if `$dirdabradio` doesn't exist.)

**Reply by klassikmann** [reply]
First test positive.

**Reply by AndyFx12** [reply]
Fix tested on 2 installations: Library update works fine, now.

**Reply by bTronics** [reply]
Fix is ok. Thanks!
